src/main/java/models/PlayerFactory.java

```diff
@@ -0,0 +1,16 @@
+package models;
+import java.util.HashMap;
+import java.util.Map;
+
+public class PlayerFactory {
+    static Map<String, PlayerType> playerTypes = new HashMap<>();
+
+    public static PlayerType getPlayerType(String name, int score) {

```

Comment:
This gets the `PlayerType` but you also have to give it a score to build the `PlayerType` with in case it didnt exist? That seems a bit weird. Why not just keep a `Player` class with a name and score?